extensions/pkg/controller/dnsrecord/reconciler.go

```diff
@@ -235,3 +240,46 @@ func (r *reconciler) delete(ctx context.Context, dns *extensionsv1alpha1.DNSReco
 
 	return reconcile.Result{}, nil
 }
+
+func updateCreatedCondition(status extensionsv1alpha1.Status, conditionStatus gardencorev1beta1.ConditionStatus, reason, message string, updateIfExisting bool) error {
+	conditions := status.GetConditions()
+	c := gardencorev1beta1helper.GetCondition(conditions, extensionsv1alpha1.ConditionTypeCreated)
+	if c != nil && !updateIfExisting {
+		return nil
+	}
+	if c != nil && c.Status == conditionStatus {
+		return nil
+	}
+	if c == nil {
+		conditions = append(conditions, gardencorev1beta1helper.InitCondition(extensionsv1alpha1.ConditionTypeCreated))
+		c = &conditions[len(conditions)-1]
+	}
+
+	builder, err := gardencorev1beta1helper.NewConditionBuilder(extensionsv1alpha1.ConditionTypeCreated)
+	if err != nil {
+		return err
+	}
+
+	*c, _ = builder.WithOldCondition(*c).WithStatus(conditionStatus).WithReason(reason).WithMessage(message).Build()
```

Comment:
Does this really work in case the condition exists, i.e. when condition is set from `false -> true`? IIUC, `gardencorev1beta1helper.GetCondition` returns a copy of the condition and overwriting the content of `*c` won't change the condition in the `conditions` array, right?

Reply:
A unit test would clarify that for sure 😉 (and even ensure that it doesn't break in the future)
Side-note: an integration for the DNSRecord controller would also be nice in general.

Reply:
@timuthy  Good catch! 
Added unit test.